frontend/src/metabase/models/components/ModelDetailPage/ModelActionDetails/ModelActionDetails.tsx

```diff
@@ -82,36 +72,24 @@ function ModelActionDetails({
     ];
   }, [handleEnableImplicitActions]);
 
-  const handleEditAction = useCallback(
-    (action: WritebackAction) => {
-      setEditingActionId(action.id);
-      showActionCreator();
-    },
-    [showActionCreator],
-  );
-
-  const handleCloseActionCreator = useCallback(() => {
-    hideActionCreator();
-    setEditingActionId(undefined);
-  }, [hideActionCreator]);
-
   const renderActionListItem = useCallback(
     (action: WritebackAction) => {
-      const onEdit = canWrite ? () => handleEditAction(action) : undefined;
       return (
         <li key={action.id} aria-label={action.name}>
-          <ModelActionListItem action={action} onEdit={onEdit} />
+          <ModelActionListItem action={action} canWrite={canWrite} />
         </li>
       );
     },
-    [canWrite, handleEditAction],
+    [canWrite],
   );
 
+  const newActionUrl = Urls.newAction(model.card() as Card);
```

Comment:
I kind of wish we don't have to cast types, but don't know what other choice we have either.

Reply:
Migrate `metabase-lib` to use `metabase-types/api/card` :(